### PR TITLE
python3Packages.oca-port: init at 0.13

### PR DESCRIPTION
**Pull request by yajo**
###### Description of changes
Useful CLI tool for [OCA](https://odoo-community.org/) modules maintenance.

@moduon MT-2586

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/oca-port/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  moduleName = "oca_port";
+in
+  buildPythonPackage rec {
+    pname = "oca-port";
+    version = "0.12";
+
+    src = fetchPypi {
+      inherit version;
+      pname = moduleName;
+      sha256 = "sha256-adwqnhe5BLo5tsjaLb0XHlTFxSKNutsb6hlTXjA4Wio=";
```
Comment (natsukium) [inline pkgs/development/python-modules/oca-port/default.nix:21]:
```suggestion
      hash = "sha256-adwqnhe5BLo5tsjaLb0XHlTFxSKNutsb6hlTXjA4Wio=";
```
The SRI hash is preferred.

File: pkgs/development/python-modules/oca-port/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      click
+      GitPython
+      requests
+    ];
+    format = "pyproject";
+    SETUPTOOLS_SCM_PRETEND_VERSION = version;
+
+    passthru.updateScript = nix-update-script { };
+
+    pythonImportsCheck = [moduleName];
```
Comment (natsukium) [inline pkgs/development/python-modules/oca-port/default.nix:37]:
```suggestion
    pythonImportsCheck = [ moduleName ];
```

File: pkgs/development/python-modules/oca-port/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    meta = with lib; {
+      description = "Tool helping to port an addon or missing commits of an addon from one branch to another.";
```
Comment (natsukium) [inline pkgs/development/python-modules/oca-port/default.nix:40]:
```suggestion
      description = "Tool helping to port an addon or missing commits of an addon from one branch to another";
```

File: pkgs/development/python-modules/oca-port/default.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      homepage = "https://github.com/OCA/oca-port";
+      license = licenses.lgpl3;
```
Comment (natsukium) [inline pkgs/development/python-modules/oca-port/default.nix:42]:
```suggestion
      license = licenses.lgpl3Plus;
```

File: pkgs/development/python-modules/oca-port/default.nix [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      maintainers = [maintainers.yajo];
```
Comment (natsukium) [inline pkgs/development/python-modules/oca-port/default.nix:43]:
```suggestion
      maintainers = [ maintainers.yajo ];
```

File: pkgs/development/python-modules/oca-port/default.nix [hunk 6]
```diff
@@ -0,0 +1,45 @@
+{
+  buildPythonPackage,
+  click,
+  fetchPypi,
+  GitPython,
```
Comment (natsukium) [inline pkgs/development/python-modules/oca-port/default.nix:5]: gitpython?

**Review by natsukium — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/oca-port/default.nix (hunk 6 above)
Comment (yajo) [reply pkgs/development/python-modules/oca-port/default.nix:5]: It's a dependency: https://github.com/OCA/oca-port/blob/cebdc50f06268500783ec5667d6780e6a549d3a9/pyproject.toml#L18

**Comment by yajo** [comment]
All attended, thanks! I forgot to run nixpkgs-fmt over the file. It should be fine now.

File: pkgs/development/python-modules/oca-port/default.nix (hunk 6 above)
Comment (natsukium) [reply pkgs/development/python-modules/oca-port/default.nix]: Yes, GitPython is `gitpython` in nixpkgs.
This causes the build to fail.

Comment (yajo) [reply pkgs/development/python-modules/oca-port/default.nix]: Ah sorry, it was building locally and that's why it was confusing to me. Changed now.

**Comment by yajo** [comment]
AFAICS it all should be fixed now.

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 224626` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.oca-port</li>
    <li>python310Packages.oca-port.dist</li>
    <li>python311Packages.oca-port</li>
    <li>python311Packages.oca-port.dist</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM, thanks!

**Comment by yajo** [comment]
So, could we merge?

File: pkgs/development/python-modules/oca-port/default.nix [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  SETUPTOOLS_SCM_PRETEND_VERSION = version;
+
+  passthru.updateScript = nix-update-script { };
+
+  pythonImportsCheck = [ moduleName ];
+
+  meta = with lib; {
+    description = "Tool helping to port an addon or missing commits of an addon from one branch to another";
+    homepage = "https://github.com/OCA/oca-port";
+    license = licenses.lgpl3Plus;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/oca-port/default.nix:42]: meta.maintainers must be set

File: pkgs/development/python-modules/oca-port/default.nix [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+, lib
+, nix-update-script
+, requests
+, setuptools-scm
+, setuptools
+}:
+let
+  moduleName = "oca_port";
+in
+buildPythonPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/oca-port/default.nix:14]:
```suggestion
}:

buildPythonPackage rec {
```

File: pkgs/development/python-modules/oca-port/default.nix [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    click
+    gitpython
+    requests
+  ];
+  format = "pyproject";
+  SETUPTOOLS_SCM_PRETEND_VERSION = version;
+
+  passthru.updateScript = nix-update-script { };
+
+  pythonImportsCheck = [ moduleName ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/oca-port/default.nix:37]:
```suggestion
  pythonImportsCheck = [ "oca_port" ];
```

File: pkgs/development/python-modules/oca-port/default.nix [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+let
+  moduleName = "oca_port";
+in
+buildPythonPackage rec {
+  pname = "oca-port";
+  version = "0.12";
+
+  src = fetchPypi {
+    inherit version;
+    pname = moduleName;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/oca-port/default.nix:20]:
```suggestion
    pname = "oca_port",
    inherit version;
```

File: pkgs/development/python-modules/oca-port/default.nix [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+, requests
+, setuptools-scm
+, setuptools
+}:
+let
+  moduleName = "oca_port";
+in
+buildPythonPackage rec {
+  pname = "oca-port";
+  version = "0.12";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/oca-port/default.nix:16]:
```suggestion
  version = "0.12";
  format = "pyproject";
```

File: pkgs/development/python-modules/oca-port/default.nix [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+    setuptools
+    setuptools-scm
+  ];
+  propagatedBuildInputs = [
+    click
+    gitpython
+    requests
+  ];
+  format = "pyproject";
+  SETUPTOOLS_SCM_PRETEND_VERSION = version;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/oca-port/default.nix:33]:
```suggestion
    hash = "sha256-adwqnhe5BLo5tsjaLb0XHlTFxSKNutsb6hlTXjA4Wio=";
  };

  nativeBuildInputs = [
    setuptools
    setuptools-scm
  ];

  propagatedBuildInputs = [
    click
    gitpython
    requests
  ];

  SETUPTOOLS_SCM_PRETEND_VERSION = version;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by yajo** [comment]
All attended.

**Comment by SuperSandro2000** [comment]
@ofborg build python310Packages.oca-port

**Pull request merged**